src/Microsoft.ML.ImageAnalytics/VectorToImageTransform.cs

```diff
@@ -114,23 +125,274 @@ public class Arguments : TransformInputBase
+                _columns[i].Save(ctx);
+        }
+
+        private static (string outputColumnName, string inputColumnName)[] GetColumnPairs(VectorToImageConvertingEstimator.ColumnInfo[] columns)
+        {
+            Contracts.CheckValue(columns, nameof(columns));
+            return columns.Select(x => (x.Name, x.InputColumnName)).ToArray();
+        }
+
+        private protected override IRowMapper MakeRowMapper(DataViewSchema schema) => new Mapper(this, schema);
+
+        protected override void CheckInputColumn(DataViewSchema inputSchema, int col, int srcCol)
+        {
+            var inputColName = _columns[col].InputColumnName;
+            var vectorType = inputSchema[srcCol].Type as VectorType;
+            if (vectorType == null)
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", inputColName, "image", inputSchema[srcCol].Type.ToString());
+
+            if (vectorType.GetValueCount() != _columns[col].Height * _columns[col].Width * _columns[col].Planes)
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", inputColName, "known-size image", "unknown-size image");
+        }
+
+        private sealed class Mapper : OneToOneMapperBase
+        {
+            private readonly VectorToImageConvertingTransformer _parent;
+            private readonly ImageType[] _types;
+
+            public Mapper(VectorToImageConvertingTransformer parent, DataViewSchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _types = ConstructTypes(_parent._columns);
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+                => _parent._columns.Select((x, idx) => new DataViewSchema.DetachedColumn(x.Name, _types[idx], null)).ToArray();
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Host.AssertValue(input);
+                Host.Assert(0 <= iinfo && iinfo < _parent._columns.Length);
+
+                var type = _types[iinfo];
+                var ex = _parent._columns[iinfo];
+                bool needScale = ex.Offset != 0 || ex.Scale != 1;
+                disposer = null;
+                var sourceType = InputSchema[ColMapNewToOld[iinfo]].Type;
+                var sourceItemType = sourceType.GetItemType();
+                if (sourceItemType == NumberDataViewType.Single || sourceItemType == NumberDataViewType.Double)
+                    return GetterFromType<float>(input, iinfo, ex, needScale);
+                else
+                    if (sourceItemType == NumberDataViewType.Byte)
+                    return GetterFromType<byte>(input, iinfo, ex, false);
+                else
+                    throw Contracts.Except("We only support float or byte arrays");
```

Comment:
>We only support float or byte arrays [](start = 44, length = 36)

From estimator: `"known-size vector of type float, double or byte`
Double get lost.